source/MapPanel.h

```diff
@@ -125,7 +125,7 @@ class MapPanel : public Panel {
 	// for use in determining which governments are in the legend.
 	std::map<const Government *, double> closeGovernments;
 	// Systems in which your (active and parked) escorts and stored outfits are located.
-	std::map<const System *, std::pair<std::pair<int, int>, int>> escortSystems;
+	std::map<const System *, std::pair<std::pair<int, int>, std::map<const Planet *, unsigned>>> escortSystems;
```

Comment:
If we keep this tooltip at the system level, I think it makes sense to introduce a nested class, e.g.

```c++
class SystemTooltipData {
public:
  unsigned parkedShips = 0;
  unsigned activeShips = 0;
  std::map<const Planet *, unsigned> outfits;
}
```
Then this field is just `std::map<const System *, SystemTooltipData>` instead of some unwieldy mess of `.first`s and `.second`s that is only comprehensible inside an IDE.

Reply:
As part of this PR, or as a separate PR?

Reply:
It would make sense to add a `SystemTooltipData` class when adding a `PlanetTooltipData` class. I think we all can agree that a map of nested pairs is not readable without IDE features.

Reply:
Yes, the nested classes look a lot more readable. The question from jostephd is however not yet answered, so hereby my view on that;
The readability of the affected code was not good before thisPR, but also gets a little worse by the change in this PR, so the addition of the nested classes should be done in a separate PR that gets merged earlier than this PR (which would give you a merge-conflict for this PR), or that should be addressed inside this PR (which would make this PR a little bit bigger).
I feel that the change is not too large, so combining with this PR is probably preferred from an Changelog/review perspective (but if you feel that the change is sufficiently large to deserve it's own PR, then please feel free to choose a separate PR instead.)

Reply:
Thanks, @petervdmeer. From a review perspective, I think the nested class refactoring should be a separate commit _after_ merging to master, which means it should be a separate PR, because this project squashes PRs when merging them. So I'll do a separate PR. But I'd prefer to do it as a _follow-up_ PR so I don't have to resolve conflicts, please. (I'm perfectly capable of rebasing and resolving conflicts, but it takes time, which I don't have an abundance of.)

Or if you prefer, I can squash this PR to two commits (one for the tooltips changes and one for the pairs/class refactoring) so it can be merged by rebasing rather than by squashing.

Reply:
Good considerations, but please keep in mind that me and most other reviewers also don't have an abundance of time available. If your approach would result in a PR that is still efficiently reviewable, then I don't see much issues with your approach. But if your approach would make this PR (more) difficult to review, then that could become a bottleneck to get this PR towards merging.

I don't have permissions to merge PRs on this repository, but I feel that you are better of creating 2 separate PRs if you feel strongly about the commits needing to be visible separately on master. (Rebase merges will take a little bit more effort from the devs and the devs already seem to be quite busy.)

Reply:
It makes sense to me to implement both tooltip classes in the same PR. That said, if implementing orbits-based tooltips wouldn't be touching any of the same code, splitting it could make sense as there won't be much in the way of conflicts.

W.r.t. dealing with squash merges of a precursor feature branch, I've utilized [`git post-squash`](https://github.com/nomeata/git-post-squash) at work a few times to good effect.

Reply:
> Good considerations, but please keep in mind that me and most other reviewers also don't have an abundance of time available.

Of course. I am not trying to save time for myself at the reviewers' expense. On the contrary, I proposed what, in my opinion, would _save_ time for reviewers.

> If your approach would result in a PR that is still efficiently reviewable, then I don't see much issues with your approach. But if your approach would make this PR (more) difficult to review, then that could become a bottleneck to get this PR towards merging.

Personally, I think putting the refactoring and the tooltip-adding in the same PR, and then squash-merging it, would make the resulting commit on master more difficult to review than if the two parts were separate commits on master. That's why I proposed what I did. However, it's not my call. If y'all prefer that both changes be made in the same PR, I can do that.

> It makes sense to me to implement both tooltip classes in the same PR. That said, if implementing orbits-based tooltips wouldn't be touching any of the same code, splitting it could make sense as there won't be much in the way of conflicts.

Implementing everything in the same PR feels to me a little like making the perfect the enemy of the good. Consider, there are several pieces to this puzzle:

1. Convert the nested pairs to a class.

2. Add systems tooltips.

3. Add orbits tooltips.

This PR does (2), and that part is basically ready (except that https://github.com/endless-sky/endless-sky/pull/5942#pullrequestreview-659178759 is outstanding). This PR doesn't add the new feature in (3), and neither improves on the existing technical debt in (1) nor regresses it. So if it were up to me, I'd have considered merging this PR as soon as https://github.com/endless-sky/endless-sky/pull/5942#pullrequestreview-659178759 were resolved. I mean, it's not like adding the systems tooltip before the other parts has any obvious downside, not that I can see, anyway, regardless of whether orbits tooltips do or don't touch the same code as the other parts.

Anyway, I've made my point. Please let me know how you'd like to proceed, whether I should add (1) and (3) to this PR, or whether this PR can be merged first and (1) and (3) done subsequently.

> W.r.t. dealing with squash merges of a precursor feature branch, I've utilized [`git post-squash`](https://github.com/nomeata/git-post-squash) at work a few times to good effect.

Thanks, but it sounds like a plain `git rebase --onto=foo bar baz` could do it just as well.

Reply:
We're in a code freeze currently, waiting for a Steam release. No timeline on when it will lift, as MZ has time for the release whenever he has time for the release.

So if one approach lets you use your current free time to good effect, that's another argument for it.

Reply:
@jostephd You seem to have thought about this well. I also made my point and I don't have a strong preference, please feel free to choose the approach you feel is best.

Reply:
@tehhowch @petervdmeer Alright. I've added the refactoring to this PR, and partially squashed this PR so the refactoring is a separate commit from the tooltip adding. Thanks for the constructive discussion.

Next on my list is to figure out the preferences issue (https://github.com/endless-sky/endless-sky/pull/5942#pullrequestreview-659178759).